.circleci/scripts/common.sh

```diff
@@ -0,0 +1,24 @@
+cleanup() {
+    pkill -P $$ || true
+    wait || true
+}
+
+trap_add() {
+    trap_add_cmd=$1; shift || fatal "${FUNCNAME} usage error"
+    for trap_add_name in "$@"; do
+        trap -- "$(
+            # helper fn to get existing trap command from output
+            # of trap -p
+            extract_trap_cmd() { printf '%s\n' "${3:-}"; }
+            # print existing trap command with newline
+            eval "extract_trap_cmd $(trap -p "${trap_add_name}")"
+            # print the new trap command
+            printf '%s\n' "${trap_add_cmd}"
+        )" "${trap_add_name}" \
+            || fatal "unable to add to trap ${trap_add_name}"
+    done
+}
+
+declare -f -t trap_add
+
+trap_add 'cleanup' EXIT
```

Comment:
why is this script needed? I'd rather not cargo cult CI files.

Reply:
It's not strictly needed. It's used to cleanly shut down oasis-chain.

Reply:
ehh who cares about clean when the vm is going to be recycled anyway